androidenhancedvideoplayer/src/main/java/com/profusion/androidenhancedvideoplayer/utils/TrackUtils.kt

```diff
@@ -0,0 +1,65 @@
+package com.profusion.androidenhancedvideoplayer.utils
+
+import androidx.annotation.OptIn
+import androidx.media3.common.C
+import androidx.media3.common.Format
+import androidx.media3.common.TrackSelectionOverride
+import androidx.media3.common.Tracks
+import androidx.media3.common.util.UnstableApi
+import androidx.media3.exoplayer.ExoPlayer
+
+private const val NO_ROLE_FLAGS = 0
+
+@OptIn(UnstableApi::class)
+fun generateTrackQualityOptions(tracks: Tracks): List<TrackQualityItem> {
+    val qualityOptions = ArrayList<TrackQualityItemValue>()
```

Comment:
Don't we need to return an immutable list to be remembered? Otherwise it won't trigger recomposition. Checking the [official docs](https://developer.android.com/jetpack/compose/state#state-in-composables):

> Caution: Using mutable objects such as ArrayList<T> or mutableListOf() as state in Compose causes your users to see incorrect or stale data in your app. Mutable objects that are not observable, such as ArrayList or a mutable data class, are not observable by Compose and don't trigger a recomposition when they change. Instead of using non-observable mutable objects, the recommendation is to use an observable data holder such as State<List<T>> and the immutable listOf().

Reply:
Won't `qualityOptions.sortedDescending()` convert that arraylist to an immutable list?

Reply:
No. `sortedDescending()` returns a `List<T>` type from its definition:

```kotlin
/**
 * Returns a list of all elements sorted descending according to their natural sort order.
 * 
 * The sort is _stable_. It means that equal elements preserve their order relative to each other after sorting.
 */
public fun <T : Comparable<T>> Iterable<T>.sortedDescending(): List<T> {
    return sortedWith(reverseOrder())
}
```

Reply:
You can use:

```kotlin
fun generateTrackQualityOptions(tracks: Tracks): ImmutableList<TrackQualityItem> {
    val qualityOptions = ArrayList<TrackQualityItemValue>()

    for (group in tracks.groups) {
        for (trackIndex in 0 until group.length) {
            if (group.isTrackSupported(trackIndex) &&
                group.mediaTrackGroup.type == C.TRACK_TYPE_VIDEO &&
                group.getTrackFormat(trackIndex).isVideoTrackPlayable()
            ) {
                qualityOptions += TrackQualityItemValue(
                    group = group.mediaTrackGroup,
                    trackIndex = trackIndex
                )
            }
        }
    }

    return ImmutableList.copyOf(qualityOptions.sortedDescending())
}
```

Reply:
Done!